File: WebApplication/src/approve-registrations/dtos/search/approve-registration-search.dto.ts
import { ApproveRegistrationFileDto } from "../approve-registration-file.dto";
import { BaseApproveRegistrationDto } from "../base/base-approve-registration.dto";
import { ApproveRegistrationHistorySearchDto } from "./approve-registration-history-search.dto";

export class ApproveRegistrationSearchDto extends BaseApproveRegistrationDto {
  file: ApproveRegistrationFileDto;
  approveRegistrationHistories: ApproveRegistrationHistorySearchDto[] = [];
}

File: Sc.Models/Dtos/ApproveRegistrations/Search/ApproveRegistrationSearchDto.cs
using Sc.Models.Dtos.ApproveRegistrations.Base;

namespace Sc.Models.Dtos.ApproveRegistrations.Search
{
    public class ApproveRegistrationSearchDto : BaseApproveRegistrationDto
    {
        public ApproveRegistrationFileDto File { get; set; }

        public List<ApproveRegistrationHistorySearchDto> ApproveRegistrationHistories { get; set; } = new List<ApproveRegistrationHistorySearchDto>();
    }
}
